pkg/virt-controller/watch/application.go

```diff
@@ -0,0 +1,129 @@
+package watch
+
+import (
+	"flag"
+	golog "log"
+	"net/http"
+	"strconv"
+
+	"github.com/emicklei/go-restful"
+	clientrest "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+
+	"k8s.io/client-go/util/workqueue"
+
+	"k8s.io/client-go/kubernetes"
+
+	kubeinformers "kubevirt.io/kubevirt/pkg/informers"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/virt-controller/rest"
+	"kubevirt.io/kubevirt/pkg/virt-controller/services"
+)
+
+var (
+	flags           Flags = Flags{}
```

Comment:
Picture all unit tests being run in goroutines. That's a valid thing we could actually do and use of globals in such a case gets messy. Initializing these vars is *not* something we do just once when unit testing is in play. One solution would look something like this:

```type ControllerConfig (
    host string
    port int
    templateService ...
)```

Then `Execute()` becomes e.g. `Execute(cfg *ControllerConfig)` You don't even need to initialize this object before passing it in, since that's what Execute does

This approach is threadsafe and shouldn't really impact any logic.